src/lib/Runner.ts

```diff
@@ -551,44 +556,57 @@ export class Runner {
-          namespace: 'lib:runner:checkWaitingLandRequests',
-          waitingRequestStatuses,
-        });
-
-        for (const landRequestStatus of waitingRequestStatuses) {
-          const landRequest = landRequestStatus.request;
-          const pullRequestId = landRequest.pullRequestId;
-          const triggererUserMode = await permissionService.getPermissionForUser(
-            landRequest.triggererAaid,
-          );
-          const isAllowedToLand = await this.isAllowedToLand(
-            pullRequestId,
-            triggererUserMode,
-            this.getQueue,
-          );
-
-          if (isAllowedToLand.errors.length === 0) {
-            if (isAllowedToLand.existingRequest) {
-              Logger.warn('Already has existing Land build', {
+        await withLock(
+          // this lock ensures we don't run the check when we're running this.next()
+          'status-transition',
+          async () => {
+            const waitingRequestStatuses = await this.queue.getStatusesForWaitingRequests();
+
+            Logger.info('Checking for waiting landrequests ready to queue', {
+              namespace: 'lib:runner:checkWaitingLandRequests',
+              waitingRequestStatuses,
+            });
+
+            for (const landRequestStatus of waitingRequestStatuses) {
+              const landRequest = landRequestStatus.request;
+              const pullRequestId = landRequest.pullRequestId;
+              const triggererUserMode = await permissionService.getPermissionForUser(
+                landRequest.triggererAaid,
+              );
+              const isAllowedToLand = await this.isAllowedToLand(
                 pullRequestId,
-                landRequestId: landRequest.id,
-                landRequestStatus,
-                namespace: 'lib:runner:checkWaitingLandRequests',
-              });
-              await landRequest.setStatus('aborted', 'Already has existing Land build');
-              continue;
+                triggererUserMode,
+                this.getQueue,
+              );
+
+              if (isAllowedToLand.errors.length === 0) {
+                if (isAllowedToLand.existingRequest) {
+                  Logger.warn('Already has existing Land build', {
+                    pullRequestId,
+                    landRequestId: landRequest.id,
+                    landRequestStatus,
+                    namespace: 'lib:runner:checkWaitingLandRequests',
+                  });
+                  await landRequest.setStatus('aborted', 'Already has existing Land build');
+                  continue;
+                }
+                const movedState = await this.moveFromWaitingToQueued(landRequestStatus);
+                if (movedState) return this.next();
+              }
             }
-            const movedState = await this.moveFromWaitingToQueued(landRequestStatus);
-            if (movedState) return this.next();
-          }
-        }
+          },
+          undefined,
```

Comment:
`export const withLock = async <T>(
  resource: string,
  fn: (lockId: Date) => Promise<T>,
  fallback: T,
  ttl: number = 60000,
)`

That's the fallback for the above function, so that if the lock is being occupied and can't be obtained, it will return the fallback instead of running the actual logic fn.